web/src/components/config_render/ConfigCheckbox.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { createRef } from "react";
+// TODO: add type checking support for react-remarkable or add a global ignore
+// @ts-ignore
+import Markdown from "react-remarkable";
+import { setOrder } from "./ConfigUtil";
+import { ConfigWrapper } from "./ConfigComponents";
+
+type Props = {
+  default: string;
+  groupName: string;
+  handleAddItem: () => void;
+  handleOnChange: (name: String, val: string) => void;
+  handleRemoveItem: () => void;
+  help_text: string;
+  hidden: boolean;
+  index: number;
+  name: string;
+  readonly: any;
+  title: string;
+  type: string;
+  value: string;
+  when: any;
+  affix: string;
+  className: string;
+  required: boolean;
+  recommended: boolean;
+};
+
+export default class ConfigCheckbox extends React.Component<Props> {
+  private checkbox = createRef<HTMLInputElement>();
+
```

Comment:
following https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forward_and_create_ref/ and 
https://medium.com/@martin_hotell/react-refs-with-typescript-a32d56c4d315

will be changed since i'll convert this out of class component in another PR.